### DISCUSSION TITLE
jq library is not working

### DISCUSSION THREAD

**Discussion by pennywise-01** [aquasecurity/trivy#7227] [Q&A] [answered]
### Question

I m following this [link](https://aquasecurity.github.io/trivy/v0.53/tutorials/integrations/aws-security-hub/). 
At this steps i got and error like this
![image](https://github.com/user-attachments/assets/947a13b3-6e0a-446d-9268-b9e0dd86878a)
![image](https://github.com/user-attachments/assets/2c4d5ee7-501a-4f98-a4a6-7bef8a37011b)

Anyway i can fix this?

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Template

### Mode

Standalone

### Operating System

ubuntu server 24.04

### Version

```bash
Version: 0.53.0
```

**Comment by DmitriyLewen** [comment]
Hello @Hafzan-250601 
Thanks for your report!

It works for me:
```bash
➜ trivy image -f template -t @contrib/asff.tpl ubuntu | jq '.Findings'
...
[
  {
    "SchemaVersion": "2018-10-08",
    "Id": "ubuntu (ubuntu 24.04)/CVE-2016-2781",
...
```

Can you share your image/report.asff file?

Regards, Dmitriy

**Reply by pennywise-01** [reply]
trivy image --format template --template trivy/contrib/asff.tpl -o report.asff --severity HIGH,CRITICAL,MEDIUM devopsapps

cat report.asff | jq '.Findings'

This is the command that i use

**Reply by pennywise-01** [reply]
Somehow the command works but new issue come in 
![image](https://github.com/user-attachments/assets/02a5350d-cf06-4177-8ad3-7071ecc62f9c)

How do i fix it?

**Reply by afdesk** [reply]
Hi @Hafzan-250601 
it looks like a really old known issue in AWS: 
https://github.com/aquasecurity/trivy/issues/2847#issuecomment-1241950351

> there are two ASFF struct: for the AWS CLI and for the AWS API. Findings need only for the last one.

**Reply by afdesk** [reply]
ASFF template for Trivy contains `Findings` field now: https://github.com/aquasecurity/trivy/blob/d76febaee107c645e864da0f4d74a8f6ae4ad232/contrib/asff.tpl#L2

so please share you `trivy/contrib/asff.tpl` file?

also maybe did you miss `@` in the path?

**Reply by pennywise-01** [reply]
Hi @afdesk 
Here is my Jenkins pipeline configurations
![image](https://github.com/user-attachments/assets/c792769c-2b35-4d5a-b1c8-6f2e8d76ffc0)

I already updated the command to add `@`. As for the asff.tpl I cloned the whole trivy repo and use the one under `/contrib`

**Reply by pennywise-01** [reply]
hi @afdesk 

I updated my command to this and got another error something about the format
![image](https://github.com/user-attachments/assets/cee3bef7-b863-46ef-90b7-3eda1f3e13a9)

Here is the new error
![image](https://github.com/user-attachments/assets/d7e216f0-3eec-4730-81ca-b663c2e6b2c6)

**Reply by afdesk** [reply]
@Hafzan-250601 unfortunately, I have no any jenkins instances now, 
could you share Trivy logs? and the next output:
```sh
$ cat report.asff | jq '.Findings'
```

the pass to asff.tpl shoulb be correct and contain `@`.
without `@` Trivy doesn't return json.
```sh
$ trivy image -f template -t /home/user/trivy/contrib/asff.tpl -o report.asff alpine:3.15
...
$ cat report.asff 
/home/user/trivy/contrib/asff.tpl
$ trivy image -f template -t @/home/user/trivy/contrib/asff.tpl -o report.asff alpine:3.15
$ cat report.asff 
{
    "Findings": [
    ]
}
```

**Reply by pennywise-01** [reply]
Hi @afdesk here is the trivy logs
![image](https://github.com/user-attachments/assets/701ce861-6ab6-4237-a493-efab95b6a909)

I ran it in my ubuntu. The output still the same as the one in my Jenkins

**Reply by DmitriyLewen** [reply, marked as answer]
Hello @Hafzan-250601 @afdesk 
I may have figured out what the problem is:
You are not overwriting the `result.asff` file.

example:
```bash
➜  trivy image -f template -t @contrib/asff.tpl ubuntu -o report.asff  

➜  cat report.asff 
{
    "Findings": [
...

➜ cat report.asff | jq '.Findings'
[
  {
...

➜  cat report.asff 
{
    "Findings": [
...

➜  cat report.asff | jq '.Findings' > report_updated.asff
➜  cat report_updated.asff 
[
  {
...

```

UUIC there is problem with overwriting input file using `jq`.
So best way is save updated result to another file and use new file in `aws`.

**Reply by afdesk** [reply]
Exactly! 
>  In order to upload via the CLI the outer wrapping must be removed being left with only the array of findings.

**Reply by pennywise-01** [reply]
Thanks!
